lib/dispatcher.rb

```diff
@@ -43,7 +43,8 @@ def self.initialize_dispatcher(cart)
     when 'parallel'
       ParallelDispatcher.new
     when 'linear'
-      if cart.ncr?
+      # @todo: dynamic dispatch for selection
+      if cart.proposal.client_data_legacy.client == "ncr"
```

Comment:
Ideally, there'd be zero "are you an instance of ncr?" checks. These should all be implicit based on dynamic dispatch, file existence, etc.